File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Imports/FormContexts/Northwind.CustomerForm.cs

namespace BasicApplication.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;
    using BasicApplication;

    public partial class CustomerForm : PrefixedContext
    {
        public CustomerForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor CustomerID { get { return ById<StringEditor>("CustomerID"); } }
        public StringEditor CompanyName { get { return ById<StringEditor>("CompanyName"); } }
        public StringEditor ContactName { get { return ById<StringEditor>("ContactName"); } }
        public StringEditor ContactTitle { get { return ById<StringEditor>("ContactTitle"); } }
        public StringEditor Address { get { return ById<StringEditor>("Address"); } }
        public StringEditor City { get { return ById<StringEditor>("City"); } }
        public StringEditor Region { get { return ById<StringEditor>("Region"); } }
        public StringEditor PostalCode { get { return ById<StringEditor>("PostalCode"); } }
        public StringEditor Country { get { return ById<StringEditor>("Country"); } }
        public StringEditor Phone { get { return ById<StringEditor>("Phone"); } }
        public StringEditor Fax { get { return ById<StringEditor>("Fax"); } }
    }
}

File: common-features/src/demo.northwind/Modules/ServerTypes/Demo/Northwind.CustomerForm.ts
import { BooleanEditor, DateEditor, EmailAddressEditor, initFormType, LookupEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";
import { NotesEditor } from "../../Note/NotesEditor";

export interface CustomerForm {
    CustomerID: StringEditor;
    CompanyName: StringEditor;
    ContactName: StringEditor;
    ContactTitle: StringEditor;
    Representatives: LookupEditor;
    Address: StringEditor;
    Country: LookupEditor;
    City: LookupEditor;
    Region: StringEditor;
    PostalCode: StringEditor;
    Phone: StringEditor;
    Fax: StringEditor;
    NoteList: NotesEditor;
    LastContactDate: DateEditor;
    LastContactedBy: LookupEditor;
    Email: EmailAddressEditor;
    SendBulletin: BooleanEditor;
}

export class CustomerForm extends PrefixedContext {
    static readonly formKey = 'Northwind.Customer';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!CustomerForm.init) {
            CustomerForm.init = true;

            initFormType(CustomerForm, [
                'CustomerID', StringEditor,
                'CompanyName', StringEditor,
                'ContactName', StringEditor,
                'ContactTitle', StringEditor,
                'Representatives', LookupEditor,
                'Address', StringEditor,
                'Country', LookupEditor,
                'City', LookupEditor,
                'Region', StringEditor,
                'PostalCode', StringEditor,
                'Phone', StringEditor,
                'Fax', StringEditor,
                'NoteList', NotesEditor,
                'LastContactDate', DateEditor,
                'LastContactedBy', LookupEditor,
                'Email', EmailAddressEditor,
                'SendBulletin', BooleanEditor
            ]);
        }
    }
}